### DISCUSSION TITLE
How to filter sentry messages generated from HTTPException, base on the status code (e.g. not send 422)

### DISCUSSION THREAD

**Discussion by johannamelly** [fastapi/fastapi#10577] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

@app.get("/")
def return_id(my_id: int):
     return my_id
```

### Problem
We receive our backend errors on Sentry. However, we would like to filter out the errors that are sent using their status_code.
We found 2 promising leads in sentry-python, which are `before_send` function that allows to alter the events before sending them, and `traces_sampler` argument in the `sentry_sdk.init`, which allows us to filter some events.

However, the status_code seems to appear nowhere: we checked on Sentry and in the code, and the error events that are formed do not seem to pass this information.

### Expected behavior
The event caught by sentry contains a status_code. The error is sent as normal, but as I have the status_code, I can drop the sentry event.

Another solution could be to configure from fastAPI what HTTPExceptions are sent to sentry, but we could not find how to achieve this neither.

### How to reproduce
_(with sentry configured)_
Call the endpoint, pass a string as `my_id` instead of an int. In Sentry, you get an error. There's no way of telling it's a 422.

### Operating System

macOS

### FastAPI Version

1.103.2

### Pydantic Version

1.10.13

### Python Version

3.8.12

### Additional Context

sentry_sdk  version: 1.32.0

**Comment by jgould22** [comment, marked as answer]
I am not well versed in sentry's sdk but the docs seem to indicate that[ you should be able to access the original exception](https://docs.sentry.io/platforms/python/configuration/filtering/#using-hints) in the `hint` param which should contain the status code

What does your `hint` param look like for a sample request?
